activesupport/test/callbacks_test.rb

```diff
@@ -781,13 +779,11 @@ def setup
       ActiveSupport::Callbacks::CallbackChain.halt_and_display_warning_on_return_false = true
     end
 
-    def test_returning_false_halts_callback_if_config_variable_is_true
+    def test_returning_false_does_not_halt_callback_if_config_variable_is_true
       obj = CallbackFalseTerminator.new
-      assert_deprecated do
-        obj.save
-        assert_equal :second, obj.halted
-        assert !obj.saved
-      end
+      obj.save
+      assert_equal nil, obj.halted
+      assert obj.saved
```

Comment:
Exactly. 

In Rails 5 you halt the callback chain by throwing `:abort`. If you return false, the chain is not halted and the object is saved